pkg/ingester/ingester_v2.go

```diff
@@ -460,13 +503,27 @@ func (i *Ingester) v2Push(ctx context.Context, req *client.WriteRequest) (*clien
 				}
 			}
 
+			cause := errors.Cause(err)
+			if cause == storage.ErrOutOfBounds &&
+				i.cfg.BlocksStorageConfig.TSDB.BackfillLimit > 0 &&
+				s.TimestampMs > db.Head().MaxTime()-i.cfg.BlocksStorageConfig.TSDB.BackfillLimit.Milliseconds() {
```

Comment:
1. There could be any case when `db.Head().MaxTime()` `math.MinInt64`?
2. But should be based on head max time or `time.Now()`? The "max age" could be also see an compared to "now". All other time-based limits we have are based on "now".